File: csharp/difference-of-squares/Squares.cs
using System;
using System.Linq;

namespace Exercism
{
    public class Squares
    {
        private readonly int number;

        public Squares(int number)
        {
            if (number < 0)
            {
                throw new ArgumentException("Negative numbers are not allowed.", "number");
            }

            this.number = number;
        }

        public int SumOfSquares()
        {
            return Enumerable.Range(1, this.number).Sum(i => i * i);
        }

        public int SquareOfSums()
        {
            return (int)Math.Pow(Enumerable.Range(1, this.number).Sum(), 2);
        }

        public int DifferenceOfSquares()
        {
            return this.SquareOfSums() - this.SumOfSquares();
        }
    }
}

File: kotlin/difference-of-squares/src/main/kotlin/Squares.kt
class Squares(private val number: Int) {
    fun squareOfSum() = 1.rangeTo(number).sum().square()
    fun sumOfSquares() = 1.rangeTo(number).sumBy { it.square() }
    fun difference() = squareOfSum() - sumOfSquares()

    private fun Int.square() = this * this
}